google/cloud/internal/status_utils.cc

```diff
@@ -20,11 +20,13 @@ namespace cloud {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
 
-bool IsClientOrigin(Status const& status) {
+ErrorOrigin GetErrorOrigin(Status const& status) {
```

Comment:
comment: I think returning a `bool` was fine for readability.

```cc
if (IsClientOrigin(status)) {
  GetRequestMetadata(context, kWithoutHeaders);
}
```

Returning a `bool` is a poor choice when the function name doesn't give you any hints. e.g.: https://github.com/googleapis/google-cloud-cpp/blob/a77950e488c3be12832aa5efece7a0889ff47a08/google/cloud/retry_policy.h#L61

... where I have to look every time whether `true` means retry, or `false` means retry.

Reply:
I see that you are trying to unify two separate things, but I don't think it makes sense. If we need two enums, one for `ErrorOrigin` and one for `GrpcMetadataView`, so be it.